java-frontend/src/test/java/org/sonar/java/se/JavaCheckVerifier.java

```diff
@@ -371,6 +367,12 @@ private static void assertAttributeMatch(Object value, Map<Expectations.IssueAtt
     }
   }
 
+  private static void softAssertAttributeMatch(Object value, Map<Expectations.IssueAttribute, Object> attributes, Expectations.IssueAttribute attribute, SoftAssertions softAssertions) {
+    if (attributes.containsKey(attribute)) {
+      softAssertions.assertThat(value).as("attribute mismatch for " + attribute + ": " + attributes).isEqualTo(attribute.getter.apply(attributes.get(attribute)));
+    }
+  }
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a call to 'assertAll' after all 'assertThat'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2970)